stage1_fly/run/main.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	stage1commontypes "github.com/coreos/rkt/stage1_common/types"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+
+	"github.com/coreos/rkt/common"
+	"github.com/coreos/rkt/pkg/sys"
+)
+
+const (
+	flavor = "fly"
+)
+
+type flyMount struct {
+	HostPath         string
+	TargetPrefixPath string
+	RelTargetPath    string
+	Fs               string
+	Flags            uintptr
+}
+
+type volumeMountTuple struct {
+	V types.Volume
+	M schema.Mount
+}
+
+var (
+	debug        bool
+	netList      common.NetList
+	interactive  bool
+	privateUsers string
+	mdsToken     string
+	localhostIP  net.IP
+	localConfig  string
+)
+
+func getHostMounts() (map[string]struct{}, error) {
+	hostMounts := map[string]struct{}{}
+
+	mi, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return nil, err
+	}
+	defer mi.Close()
+
+	sc := bufio.NewScanner(mi)
+	for sc.Scan() {
+		var (
+			discard    string
+			mountPoint string
+		)
+
+		_, err := fmt.Sscanf(sc.Text(),
+			"%s %s %s %s %s",
+			&discard, &discard, &discard, &discard, &mountPoint,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		hostMounts[mountPoint] = struct{}{}
+	}
+	if sc.Err() != nil {
+		return nil, fmt.Errorf("problem parsing mountinfo: %v", sc.Err())
+	}
+	return hostMounts, nil
+}
+
+func init() {
+	flag.BoolVar(&debug, "debug", false, "Run in debug mode")
+	flag.Var(&netList, "net", "Setup networking")
+	flag.BoolVar(&interactive, "interactive", false, "The pod is interactive")
+	flag.StringVar(&privateUsers, "private-users", "", "Run within user namespace. Can be set to [=UIDBASE[:NUIDS]]")
```

Comment:
If we keep this parameter, we should at least return an error when it is used...